packages/core/core/src/factories/useForgotPasswordFactory.ts

```diff
@@ -0,0 +1,69 @@
+import { CustomQuery, Context, FactoryParams, UseForgotPasswordErrors, UseForgotPassword } from '../types';
+import { Ref, computed } from '@vue/composition-api';
+import { sharedRef, Logger, configureFactoryParams } from '../utils';
+
+interface SetNewPasswordParams {
+  tokenValue: string;
+  newPassword: string;
+}
+
+interface ResetPasswordParams {
+  email: string;
+}
+
+export interface UseForgotPasswordFactoryParams<RESULT> extends FactoryParams {
+  resetPassword: (context: Context, params: ResetPasswordParams & { customQuery?: CustomQuery }) => Promise<RESULT>;
+  setNewPassword: (context: Context, params: SetNewPasswordParams & { customQuery?: CustomQuery }) => Promise<RESULT>;
+}
+
+export function useForgotPasswordFactory<RESULT>(
+  factoryParams: UseForgotPasswordFactoryParams<RESULT>
+) {
+  return function useForgotPassword(): UseForgotPassword<RESULT> {
+    const result: Ref<RESULT> = sharedRef({}, 'useForgotPassword-result');
+    const loading = sharedRef(false, 'useProduct-loading');
+    const _factoryParams = configureFactoryParams(factoryParams);
+    const error: Ref<UseForgotPasswordErrors> = sharedRef({
+      request: null,
+      setNew: null
+    }, 'useForgotPassword-error');
+
+    const request = async (resetPasswordParams: ResetPasswordParams) => {
+      Logger.debug('useForgotPassword/request', resetPasswordParams.email);
+
+      try {
+        loading.value = true;
+        result.value = await _factoryParams.resetPassword(resetPasswordParams);
+        error.value.request = null;
+      } catch (err) {
+        error.value.request = err;
+        Logger.error('useForgotPassword/request', err);
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    const setNew = async (setNewPasswordParams: SetNewPasswordParams) => {
+      Logger.debug('useForgotPassword/setNew');
```

Comment:
What about `setNewPasswordParams` in Logger.debug? I bet you didn't want to display it - However, after I wondered about it - It will be only called client side and with certain logging level, you can read only your own password. It might be useful to have it there and not dangerous imo